### DISCUSSION TITLE
Feature Request: Extend Python Expression API to support lambda functions

### DISCUSSION THREAD

**Discussion by binste** [duckdb/duckdb#14696] [Ideas]
While building out the Spark API, I noticed that I can't write lambda functions with the Python Expression API. I think this is needed to implement [array_compact](https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.functions.array_compact.html), [array_remove](https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.functions.array_remove.html) (both could be implemented with `list_filter` and an appropriate lambda function) and it would simplify the implementation of [arrays_overlap](arrays_overlap) (#14620) and would enable the `null_replacement` argument in [array_join](https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.functions.array_join.html) (#14620).

Maybe something like `LambdaExpression(("x", "y"), "x + y")` or an even more generic `SQLExpression("x, y -> x + y")` which could be even more flexible? No idea how much work this is and if it's possible.

**Comment by Tishj** [comment]
I can look into that at some point, I expect it to not be difficult, just requires some glue

**Comment by binste** [comment]
Thanks again! Closing it as per https://github.com/duckdb/duckdb/pull/14713
